### DISCUSSION TITLE
Sending .wav files

### DISCUSSION THREAD

**Discussion by aranis22** [pschatzmann/ESP32-A2DP#305] [Q&A] [answered]
Hi I'm pretty new to ESPs and embedded systems in general. 
In a nutshell:
I'm trying to implement a way to send WAV files from an esp32 to a PC or another esp32. The end goal is to run a code through the wav files in real time. I've already made the project on an arduino nano and it works by sending the wav files to and SD card connected to it. If I could save the hassle of having to send the file through the SD card, that would be amazing. I've thought about some ways of implementing this, including MQTT, wifi, Bluetooth, or UDP. The problem is I'm not sure how they work and which one I should use for my case. The files will be around 2mb, and I need them (ideally) to be sent in real time. I'm not sure on how to use your code to send the files through.
If you have any suggestions, please let me know

**Comment by pschatzmann** [comment, marked as answer]
__Sending real time audio via the network is hard__ and depending on your selected sample rate you operate at (or over) the limit what a microcontroller can process.  

If you are new to microcontrollers (or the ESP32) this is definitely not what you should pick as your first project. 

If you decide to continue anyhow here are a couple of examples to get you started:
- https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-communication
- or look at the A2DP examples

In any case start with small steps and measure your expected thruput with what you want to achieve..

This might help as well:
- https://github.com/pschatzmann/arduino-audio-tools/wiki/It's-not-working
- https://github.com/pschatzmann/arduino-audio-tools/wiki/Encoding-and-Decoding-of-Audio
